src/cpuinfo_x86.c

```diff
@@ -1330,6 +1330,44 @@ static void ParseCpuId(const uint32_t max_cpuid_leaf, X86Info* info,
     features->ssse3 = GetDarwinSysCtlByName("hw.optional.supplementalsse3");
     features->sse4_1 = GetDarwinSysCtlByName("hw.optional.sse4_1");
     features->sse4_2 = GetDarwinSysCtlByName("hw.optional.sse4_2");
+#elif defined(CPU_FEATURES_OS_FREEBSD)
+    // Handling FreeBSD platform through parsing /var/run/dmesg.boot.
+    const int fd = CpuFeatures_OpenFile("/var/run/dmesg.boot");
+    if (fd >= 0) {
+      StackLineReader reader;
+      StackLineReader_Initialize(&reader, fd);
+      for (;;) {
```

Comment:
is there any difference with using `while (true) {` ?

Reply:
No difference between `for(;;)` and `while(true)`. We could use either one.